packages/lib-sourcify/src/lib/verification.ts

```diff
@@ -506,6 +513,20 @@ export function addLibraryAddresses(
   };
 }
 
+export function checkCallProtectionAndReplaceAddress(
+  template: string,
+  real: string
+): string {
+  const push20CodeOp = '73';
+  const callProtection = `0x${push20CodeOp}${'00'.repeat(20)}`;
+
+  if (template.startsWith(callProtection)) {
+    const replacedCallProtection = real.slice(0, 0 + callProtection.length);
+    return replacedCallProtection + template.substring(callProtection.length);
+  }
+  return template;
+}
+
```

Comment:
Doesn't this code only check the beginning of the bytecode with `.startsWith()`? 

How is the `viaIR` case passing? Does the `viaIR` case here have `0x73` at the beginning of the deployed bytecode? We should handle the case when the call protection is found in the middle of the bytecode.

Reply:
no, it handles it via the `immutableReferences`.

Reply:
Ahh right